datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSActions.java

```diff
@@ -64,9 +75,20 @@ protected void doExecute(SdxContext context, StartUpdateLoadBalancerDNSEvent pay
                 updateLoadBalancerDNSService.performLoadBalancerDNSUpdate(sdxCluster);
                 LOGGER.info("Successfully updated the load balancer DNS for cluster {}.", sdxCluster.getClusterName());
                 eventSenderService.notifyEvent(sdxCluster, context, ResourceEvent.UPDATE_LOAD_BALANCER_DNS_FINISHED);
+                sendNotificationInCaseOfResizeRecovery(context, sdxCluster);
                 sendEvent(context, UPDATE_LOAD_BALANCER_DNS_SUCCESS_EVENT.event(), payload);
             }
 
+            private void sendNotificationInCaseOfResizeRecovery(SdxContext context, SdxCluster sdxCluster) {
+                Optional<FlowLogWithoutPayload> lastFlowLog = flowLogService.getLastFlowLog(context.getFlowParameters().getFlowId());
+                if (lastFlowLog.isPresent()) {
+                    Optional<FlowChainLog> flowChainLog = flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(lastFlowLog.get().getFlowChainId());
+                    if (flowChainLog.isPresent() && flowChainLog.get().getFlowChainType().equals(DatalakeResizeFlowEventChainFactory.class.getSimpleName())) {
+                        eventSenderService.sendEventAndNotification(sdxCluster, context.getUserId(), ResourceEvent.DATALAKE_RECOVERY_FINISHED);
+                    }
+                }
+            }
```

Comment:
Like above, I would appreciate having some UT coverage at least for the new logic changes.

Reply:
Sorry, this one seems unresolved.